### ISSUE TITLE
[BUG] Set-PnPPage not working with Page-Object but with name

### ISSUE THREAD

**Issue by HeKr09** [pnp/powershell#426]
### Reporting an Issue or Missing Feature
The Set-PnPPage doesn't seem to work with a page object but with the name.

### Expected behavior 
I'd expect to run Set-PnPPage with a variable to contains a page object as the archieved version of PnP.

### Actual behavior 
I ran the following code:
```
Set-PnPPage -Identity $page -HeaderType Custom -PromoteAs Template -ThumbnailUrl "$($placeholderImg.ServerRelativeUrl)" -ServerRelativeImageUrl "$($placeholderImg.ServerRelativeUrl)" -TranslateX 51.6301443078568 -TranslateY 36.0714285714286 -Publish | Out-Null
```
I got the following error:
```
Set-PnPPage : Insufficient arguments to update a client side page
At D:\TFS\Create-ContentTemplate.ps1:280 char:5
+     Set-PnPPage -Identity $page -HeaderType Custom -PromoteAs Templat ...
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (:) [Set-PnPPage], Exception
    + FullyQualifiedErrorId : EXCEPTION,PnP.PowerShell.Commands.Pages.SetPage
```
The following code does not throw an error (Differs in the Identity parameter).
```
Set-PnPPage -Identity $page.Name -HeaderType Custom -PromoteAs Template -ThumbnailUrl "$($placeholderImg.ServerRelativeUrl)" -ServerRelativeImageUrl "$($placeholderImg.ServerRelativeUrl)" -TranslateX 51.6301443078568 -TranslateY 36.0714285714286 -Publish | Out-Null
```

### Steps to reproduce behavior
See above

### What is the version of the Cmdlet module you are running?
1.4.0

### Which operating system/environment are you running PnP PowerShell on?
- [x] Windows
- [ ] Linux
- [ ] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by erwinvanhunen** [comment]
How did you acquire the page object?

**Comment by HeKr09** [comment]
```
$Template = 'Content Template full screen'
$page = Add-PnPPage -Name $Template -LayoutType Article -PromoteAs Template
```

**Comment by HeKr09** [comment]
Any progress here?

**Comment by erwinvanhunen** [comment]
We found a bug causing this. It will be fixed in the nightly release that is available tomorrow (1.4.15-nightly)
